File: Java/BaekJoon/Silver/S2_1541.java
package BaekJoon.Silver;

import java.util.Scanner;

public class S2_1541 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        String str = sc.nextLine();

        boolean isM = false;
        String temp = "";
        long sum1 = 0;
        long sum2 = 0;

        for (int i = 0; i < str.length(); i++) {
            char ch = str.charAt(i);

            if (ch >= '0' && ch <= '9') {
                temp += ch;
            } else if (!isM && (ch == '+' || ch == '-')) {
                if (ch == '-') isM = true;

                sum1 += Integer.parseInt(temp);
                temp = "";
            } else if (isM && (ch == '+' || ch == '-')) {
                sum2 += Integer.parseInt(temp);
                temp = "";
            }
        } 

        if (isM) {
            System.out.println(sum1 - sum2 - Integer.parseInt(temp));
        } else {
            System.out.println(sum1 + Integer.parseInt(temp));
        }

        sc.close();
    }
}

File: C/BaekJoon/Silver/S2_1541.c
#include <stdio.h>

int main() {
  char str[51];
  scanf("%s", str);

  int check_minus = 0;
  long long temp = 0;

  long long sum1 = 0;
  long long sum2 = 0;

  for (int i = 0; str[i] != '\0'; i++) {
    char ch = str[i];

    if (ch >= '0' && ch <= '9') {
      temp = temp * 10 + (int)(ch - '0');
      continue;
    }

    if (check_minus == 0 && ch == '+') {
      sum1 += temp;
      temp = 0;
      continue;
    }

    if (check_minus == 0 && ch == '-') {
      check_minus = 1;
      sum1 += temp;
      temp = 0;
      continue;
    }

    if (check_minus == 1 && (ch == '+' || ch == '-')) {
      sum2 += temp;
      temp = 0;
      continue;
    }
  }

  if (check_minus == 1) {
    printf("%lld", sum1 - sum2 - temp);
  } else {
    printf("%lld", sum1 + temp);
  }

  return 0;
}